#!/bin/bash


echo Start Job

date

echo Reordering genomes using Mauve against S. aureus NCTC8325

module load mauveAligner/4736-gompi-2020a

mkdir ./SA/01_data_analysis/03_reordered_genomes
cd ./SA/01_data_analysis/03_reordered_genomes

for genome in ./SA/00_data/03_genomes/*.fasta
do
   isolate=$(basename $genome .fasta)
   echo Reordering $isolate

   java -Xmx500m -cp /kyukon/home/gent/476/vsc47636/mauve_snapshot_2015-02-13/Mauve.jar org.gel.mauve.contigs.ContigOrderer -output $isolate -draft $genome -ref ./SA/01_data_analysis/NCTC8325.fasta

done


echo Moving last alignment for each genome to parent directory

base_dir="./SA/01_data_analysis/03_reordered_genomes"

for genome_dir in "$base_dir"/*/; do
    genome=$(basename "$genome_dir") # Get name of genome

    highest_alignment=$(find "$genome_dir" -type d -name "alignment*" | sort -V | tail -n 1) # Identify the folder of the last alignment

    fasta_file=$(find "$highest_alignment" -type f -name "$genome.fasta") # Get genome of the isolate

    cp "$fasta_file" "$base_dir" # Copy file to parent directory
    echo Genome $genome was copied

done


echo Deleting intermediate files generated by Mauve

for genome_dir in "$base_dir"/*/; do
    rm -r "$genome_dir"
done



date

echo End Job
